File: lecture_code/inflearn/querydsl/querydsl/src/main/java/study/querydsl/domain/Team.java
package study.querydsl.domain;

import jakarta.persistence.*;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

@Entity
public class Team {

    public Team() {
    }

    public Team(String name) {
        this.name = name;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @OneToMany(mappedBy = "team")
    private List<Member> members = new ArrayList<>();

    public void addMember(@NotNull Member member) {
        this.members.add(member);
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public List<Member> getMembers() {
        return members;
    }
}

File: lecture_code/inflearn/querydsl/querydsl/src/main/kotlin/study/querydsl/domain/Team.kt
package study.querydsl.domain

import jakarta.persistence.Entity
import jakarta.persistence.FetchType.LAZY
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType.IDENTITY
import jakarta.persistence.Id
import jakarta.persistence.OneToMany

//@Entity
open class Teams(
//    @Id @GeneratedValue(strategy = IDENTITY) val id: Long? = null,
    val name: String,

//    @OneToMany(mappedBy = "team", fetch = LAZY)
//    val members: MutableList<Member> = mutableListOf()
) {
//    fun addMember(member: Member) {
//        this.members.add(member)
//    }
}
